### PR TITLE
Add info on stringizers for doubles

### LINKED ISSUE #449: Describe how to stringify double values with a requested precision.
Present how to use custom stringizers to prevent confusing assertion messages when working with `double` values:

image

```cpp
#include <sstream>
#include <iomanip>

namespace snowhouse {

  template<>
  struct Stringizer<double>
  {
    static std::string ToString(double value)
    {
      std::ostringstream oss;
      oss << std::setprecision(15) << value;
      return oss.str();
    }
  };
}
```

### PR DESCRIPTION
**Pull request by hobovsky**
Resolves #449 .
Preview: [here](https://deploy-preview-456--reverent-edison-2864ea.netlify.app/languages/cpp/igloo/stringizers#precision-loss-in-formatted-double-values)

### CODE REVIEW

File: content/languages/cpp/igloo/stringizers.md [hunk 1]
```diff
@@ -247,6 +247,56 @@ should_pretty_print_vector_of_pairs_of_custom_type
...
+can result in output similar to:
+
+```text
+Test_Describe
+  Test_It
+    Expected: equal to 32.9877 (+/- 1e-09)
+    Actual: 32.9877
+```
+
+Confusingly, tests do not pass even though the actual value is presented as identical to the expected one. However, both values differ at decimal places affected by accurracy of the comparison (above `1e-9`), but below the precision of formatting. To improve presentation of `double` values in assertion messages, a specialization for `Stringizer<double>` has to be provided, preferrably in the Preloaded snippet:
```
[the remark this answers is not in the document]
Comment (hobovsky) [reply content/languages/cpp/igloo/stringizers.md:267]:
```suggestion
Confusingly, tests do not pass even though the actual value is presented as identical to the expected one. However, both values differ at decimal places affected by accurracy of the comparison (above `1e-9`), but below the precision of formatting. To improve presentation of `double` values in assertion messages, a specialization for `Stringizer<double>` has to be provided, preferably in the Preloaded snippet:
```

File: content/languages/cpp/igloo/stringizers.md [hunk 2]
```diff
@@ -247,6 +247,56 @@ should_pretty_print_vector_of_pairs_of_custom_type
...
+    {
+      std::ostringstream oss;
+      oss << std::setprecision(15) << value;
+      return oss.str();
+    }
+  };
+}
+```
+
+Precision of the formatting (in the example above the value of `1e-15` is used) can be fine tuned by authors to conform to requirements of their kata. With the specialized stringizer, `double` values are presented in assertion messages with increased precision:
```
[the remark this answers is not in the document]
Comment (hobovsky) [reply content/languages/cpp/igloo/stringizers.md:288]:
```suggestion
Precision of the formatting (in the example above the value of `1e-15` is used) can be fine-tuned by authors to conform to requirements of their kata. With the specialized stringizer, `double` values are presented in assertion messages with increased precision:
```

File: content/languages/cpp/igloo/stringizers.md [hunk 3]
```diff
@@ -69,11 +69,11 @@ Test output:
 
 Not only your custom types can be affected by this issue. Snowhouse may be not able to stringify many built-in, standard, or 3rd party types. Basically, every type which does not define its version of the output stream operator (`operator <<`) is affected, and the `std::pair` template is a very common case of such types for Codewars kata.
 
-## Solutions
+### Solutions
```
Comment (kazk) [inline content/languages/cpp/igloo/stringizers.md:72]: I think it's fine to keep it at `h2`. The sidebar looks kind weird now.

image

Current:

image

Comment (kazk) [reply content/languages/cpp/igloo/stringizers.md:72]: Maybe `Solutions` header is unnecessary?

Comment (hobovsky) [reply content/languages/cpp/igloo/stringizers.md]: Good idea. I noticed the sidebar looking worse, but did not think of removing the "Solutions" header. Now I removed it and I think it looks better.

**Review by kazk — APPROVED** [review]

**Pull request merged**